certbot-dns-linode/certbot_dns_linode/dns_linode.py

```diff
@@ -0,0 +1,70 @@
+"""DNS Authenticator for Linode."""
+import logging
+
+import zope.interface
+from lexicon.providers import linode
+
+from certbot import errors
+from certbot import interfaces
+from certbot.plugins import dns_common
+from certbot.plugins import dns_common_lexicon
+
+logger = logging.getLogger(__name__)
+
+@zope.interface.implementer(interfaces.IAuthenticator)
+@zope.interface.provider(interfaces.IPluginFactory)
+class Authenticator(dns_common.DNSAuthenticator):
+    """DNS Authenticator for Linode
+
+    This Authenticator uses the Linode API to fulfill a dns-01 challenge.
+    """
+
+    description = 'Obtain certs using a DNS TXT record (if you are using Linode for DNS).'
+
+    def __init__(self, *args, **kwargs):
+        super(Authenticator, self).__init__(*args, **kwargs)
+        self.credentials = None
+
+    @classmethod
+    def add_parser_arguments(cls, add):  # pylint: disable=arguments-differ
+        super(Authenticator, cls).add_parser_arguments(add, default_propagation_seconds=930)
```

Comment:
930 seconds was not _always_ sufficient in my testing. Perhaps I just got unlucky, but once in 5 attempts it took between 930 and 960.

Are you consistently seeing propagation times under 930 seconds? If so, I think this is fine as-is. If you see a higher time occasionally, it might be worth bumping this up. Waiting an extra 30 seconds every time is probably better than having to retry even 5% of the time.

Reply:
Linode updates it's DNS records from the dashboard (which is updated by the api) every 15 minutes (900 seconds), so I set the value to 930 to ensure all the linode nameservers are updated at this point. At this point, linode servers are guaranteed to respond with the new value. It should not take longer than this unless some DNS resolver caches the values. I'll change the value to 960 just in case.